source/math/IntRandomGenerator.ooc

```diff
@@ -15,17 +15,41 @@
 // along with this program. If not, see <http://www.gnu.org/licenses/>.
 
 use ooc-math
+import math
 import os/Time
 
-IntRandomGenerator: class {
-	_backend: FloatRandomGenerator
-	init: func ~withBackend (=_backend)
-	free: override func {
-		this _backend free()
-		super()
+IntUniformRandomGenerator: class {
+	_state, _min, _max, _range: Int
+	_shortRange: Bool
+	minimum ::= this _min
+	maximum ::= this _max
+	init: func (seed := Time microtime()) {
+		this setRange(0, Int maximumValue)
+		this _state = seed
+	}
+	init: func ~withParameters (min, max: Int, seed := Time microtime()) {
+		this setRange(min, max)
+		this _state = seed
+	}
+	_generate: func -> Int {
+		this _state = 214013 * this _state + 2531011
```

Comment:
Maybe a link to some website that explains the reason why these numbers are used here, if it can't be explained briefly. I have to admit I never worked out how PRNGs work, exactly, so for someone like me, it could help.